static/js/stream_create.js

```diff
@@ -283,35 +321,27 @@ $(function () {
     $(".subscriptions").on("submit", "#stream_creation_form", function (e) {
         e.preventDefault();
         var stream = $.trim($("#create_stream_name").val());
-        var description = $.trim($("#create_stream_description").val());
-
         var name_ok = stream_name_error.validate_for_submit(stream);
 
         if (!name_ok) {
             return;
         }
 
-        var principals = _.map(
-            $("#stream_creation_form input:checkbox[name=user]:checked"),
-            function (elem) {
-                return $(elem).val();
-            }
-        );
-
-        // You are always subscribed to streams you create.
-        principals.push(people.my_current_email());
-
-        created_stream = stream;
+        var principals = get_principals();
+        if (principals.length > 100) {
+            show_large_invites_warning(principals.length);
+        } else {
+            create_stream();
```

Comment:
It might be worth passing the `stream` and `principals` variables as arguments to `create_stream` as it would avoid an extra call to `get_principals()`